### PR TITLE
Migrate Notifications to support Android O

### PR DESCRIPTION
**Pull request by renatoalencar**
#### What's this PR do?
Upgrade compatibility libs to support Android O

#### Which issue(s) is it related to?
#153 

#### How to test:
Run with `compileSdkVersion` configured to 26 on a Android O.

### CODE REVIEW

**Comment by tanguyantoine** [comment]
we have #133 which updates to v27. any reason not to upgrade to 27 ?

**Comment by renatoalencar** [comment]
Since you should keep the major version of the Android support library the same as the `targetSdkVersion`, and so the project too. Also, the default `targetSdkVersion` for React Native apps is 23, so I think that it would be more cautious to slowly increase SDK version.

Actually, there are a discussion to move React Native Android SDK version to 26: https://github.com/facebook/react-native/issues/18095.

**Comment by tanguyantoine** [comment]
@Guichaguri thoughts ?

File: android/src/main/java/com/tanguyantoine/react/MusicControlModule.java [hunk 1]
```diff
@@ -57,6 +64,8 @@
 
     public NotificationClose notificationClose = NotificationClose.PAUSED;
 
+    public static final String CHANNEL_ID = "12min_playback_channel";
```
Comment (Guichaguri) [inline android/src/main/java/com/tanguyantoine/react/MusicControlModule.java]: Can you change the channel ID to something like "react-native-music-control"?

File: android/src/main/java/com/tanguyantoine/react/MusicControlModule.java [hunk 2]
```diff
@@ -84,6 +93,22 @@ public String getName() {
         return map;
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    private void createChannel(ReactApplicationContext context) {
+        NotificationManager
+            mNotificationManager =
+            (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
```
Comment (Guichaguri) [inline android/src/main/java/com/tanguyantoine/react/MusicControlModule.java]: There's no need to break this into 3 lines

File: android/src/main/java/com/tanguyantoine/react/MusicControlModule.java [hunk 3]
```diff
@@ -84,6 +93,22 @@ public String getName() {
...
+
+        CharSequence name = "Media playback";
+        String description = "Media playback controls";
+        int importance = NotificationManager.IMPORTANCE_LOW;
```
Comment (Guichaguri) [inline android/src/main/java/com/tanguyantoine/react/MusicControlModule.java]: There's no need to split these into variables as they're only being used once

**Comment by Guichaguri** [comment]
I agree, Google is going to enforce API level 26, and I don't think React Native is going to be updated to the latest API level. If we keep it at minimum, we won't have any problems whether they update to API 26 or update all the way to API 28.

Btw, I prefer this PR because it creates a notification channel, while #133 only updates the support lib.

File: android/src/main/java/com/tanguyantoine/react/MusicControlModule.java (hunk 1 above)
Comment (renatoalencar) [reply android/src/main/java/com/tanguyantoine/react/MusicControlModule.java]: Off course

File: android/src/main/java/com/tanguyantoine/react/MusicControlModule.java (hunk 2 above)
Comment (renatoalencar) [reply android/src/main/java/com/tanguyantoine/react/MusicControlModule.java]: Right

File: android/src/main/java/com/tanguyantoine/react/MusicControlModule.java (hunk 3 above)
Comment (renatoalencar) [reply android/src/main/java/com/tanguyantoine/react/MusicControlModule.java]: Ok, right.

**Comment by Guichaguri** [comment]
Thanks

**Comment by tanguyantoine** [comment]
Published as 0.7.0

**Pull request merged**